game-app/game-core/src/main/java/games/strategy/triplea/EngineImageLoader.java

```diff
@@ -23,8 +26,20 @@ public Image loadFrameIcon() {
    *
    * @param path Path from assets folder to image, eg: loadImage("folder-in-assets", "image.png");
    */
-  public Image loadImage(final String... path) {
-    final Path imageFilePath = createPathToImage(path);
+  public BufferedImage loadImage(final String... path) {
+    Path imageFilePath = createPathToImage(path);
+
+    if (!Files.exists(imageFilePath)) {
+      try {
```

Comment:
I need this for UI testing. Can we postopone this discussion?

Reply:
If we do not merge the UI code, then this will then be dead code. It is a risk. Let's update this as necessary when the UI code lands. I also do not think the fallback code even with UI testing is necessarily correct. On another level, if we need special fallback code to make testing work, that is only used for testing, are those tests actually exercising production code in a useful way?

Reply:
If we do not merge the UI code, then some @VisibleForTesting tags and the respective increased visibility also need to be reverted. So the state after this PR - according to our shared intentions - will be inconsistent anyway.

Being only a evening programmer for fun, I lack competence in version management. Can we sort out the whole issue soon after this PR? I have free time until Jan 9th.

Reply:
> If we do not merge the UI code, then some @VisibleForTesting tags and the respective increased visibility also need to be reverted.

Indeed. I'm not necessarily super worried and if we never clean those up it wouldn't cause that much harm.

In this example I'm more hesitant because I suspect we would not need the update here in any case.

Are you using a GUI for your 'git' commands @RaiNova , or CLI? I presume you are concerned about pushing updates to this branch to then have them conflict with another branch containing the UI tests. If so, this is a good case for `cherry-pick`. You can commit these changes here, and then switch branches to the other branch and run `cherr-pick <commit-SHA>` to apply those same changes.

Post squash-rebase merge you may see merge conflicts (probably would). If you're not able to resolve them I can offer to help you do so. Would that be a good compromise?

Alternatively, I can ignore these updates but am concerned about precedent and practice, that we might get confused by this later or even build on it during the interim, and/or neglect to remove the update later.

Reply:
I am using git from within IntelliJ. I don't know how to use github to work on multiple tripleA PRs that partially depend on each other (ideally there would be 5, so be prepared :wink:). I just accidently commited two files to this PR which should have gone into an independent documentation PT ☹️).

It's just a few lines of code. To prevent this from being bad precedent and practice: I promise to work with you on the UI tesing next (after two minor PRs, or maybe in parallel, if I work out how to do that). Can you agree to let it be for now?